richtextfx/src/main/java/org/fxmisc/richtext/skin/TextExt.java

```diff
@@ -0,0 +1,95 @@
+package org.fxmisc.richtext.skin;
+
+import com.sun.javafx.css.converters.PaintConverter;
+import javafx.beans.property.ObjectProperty;
+import javafx.css.CssMetaData;
+import javafx.css.Styleable;
+import javafx.css.StyleableObjectProperty;
+import javafx.css.StyleableProperty;
+import javafx.scene.paint.Color;
+import javafx.scene.paint.Paint;
+import javafx.scene.text.Text;
+
+import java.lang.reflect.Field;
+import java.util.ArrayList;
+import java.util.List;
+
+public class TextExt extends Text {
+
+    private ObjectProperty<Paint[]> backgroundColor = null;
```

Comment:
Nope sorry, this one has the same behavior. Under the hood it still injects a Paint array which leads to casting errors later on:

Exception in thread "JavaFX Application Thread" java.lang.ClassCastException: [Ljavafx.scene.paint.Paint; cannot be cast to javafx.scene.paint.Paint

Maybe I'm just too stupid to make it work so don't hold back if you want to try your luck.